src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicRequestToViewTransformer.java

```diff
@@ -295,7 +298,7 @@ private StaticLangApiMessageView generateRequestClass(
           schemaToParamView(
               context,
               requestBody,
-              DiscoGapicNamer.getSchemaNameAsParameter(requestBodyDef).toLowerCamel(),
+              DiscoGapicNamer.getSchemaNameAsParameter(requestBodyDef.dereference()).toLowerCamel(),
```

Comment:
What happens when the only thing you change is adding a call to `dereference()`?

Reply:
Specifically for the compute API, some methods acting on an `Address` object will have a request input type that contains both an `Address` resource object and an `address` String as part of the resource locator. Without dereferencing the `Address` resource object, the namer has very limited information and would just assign `"address"` as the name for the resource object, which would conflict with the String resource locator `address`. 

Moving the `dereference()` call to inside the `DiscoveryField` constructor to decrease the number of explicit dereference calls.